### PR TITLE
Fix Clippy lints on WASM

### LINKED ISSUE #13024: Clippy complains when building with wasm32.
While working on #13022 I noticed that running ```cargo clippy --target wasm32-unknown-unknown``` raised a bunch of warnings. Mostly [`machine applicable`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_lint_defs/enum.Applicability.html#variants) stuff and one undocumented unsafe block.

### PR DESCRIPTION
**Pull request by BD103**
# Objective

- Fixes #13024.

## Solution

- Run `cargo clippy --target wasm32-unknown-unknown` until there are no more errors.
  - I recommend reviewing one commit at a time :)

---

## Changelog

- Fixed Clippy lints for `wasm32-unknown-unknown` target.
- Updated `bevy_transform`'s `README.md`.

### CODE REVIEW

File: crates/bevy_transform/README.md [hunk 1]
```diff
@@ -1,3 +1,3 @@
 # Bevy Transform
 
-This crate is largely a 1:1 port from [legion_transform](https://github.com/AThilenius/legion_transform) (ecs: legion, math: nalgebra) to bevy (ecs: bevy_ecs, math: glam)
+This crate is largely a 1:1 port from [legion_transform](https://github.com/AThilenius/legion_transform) (ecs: `legion`, math: `nalgebra`) to bevy (ecs: `bevy_ecs`, math: `glam`)
```
Comment (james7132) [inline crates/bevy_transform/README.md:3]: I don't think this comment is all that true anymore.

File: crates/bevy_tasks/src/single_threaded_task_pool.rs [hunk 2]
```diff
@@ -106,10 +106,12 @@ impl TaskPool {
         T: Send + 'static,
     {
         let executor = &async_executor::LocalExecutor::new();
+        // TODO: Safety comment
```
Comment (james7132) [inline crates/bevy_tasks/src/single_threaded_task_pool.rs:109]: Copying the safety comment from the multithreaded version is fine.

File: crates/bevy_transform/README.md (hunk 1 above)
Comment (BD103) [reply crates/bevy_transform/README.md]: Ok, I updated it in 14a4a8f3ee34100052bcca6314299a292ce2faab. I fear my new description is too short, but there's not much to say.

File: crates/bevy_tasks/src/single_threaded_task_pool.rs (hunk 2 above)
Comment (BD103) [reply crates/bevy_tasks/src/single_threaded_task_pool.rs]: Got it! Fixed in 1e2bd8f2101efc396dcadc14514b19aef6d79ebe.

**Review by james7132 — APPROVED** [review]

**Review by afonsolage — APPROVED** [review]

**Review by mockersf — APPROVED** [review]

**Pull request merged**